devito/types/basic.py

```diff
@@ -948,7 +948,7 @@ def indexify(self, indices=None, lshift=False, subs=None):
         # Add halo shift
         shift = self._size_nodomain.left if lshift else tuple([0]*len(self.dimensions))
         # Indices after substitutions
-        indices = [(a - o + f).xreplace(s) for a, o, f, s in
+        indices = [sympy.nsimplify((a - o + f).xreplace(s)) for a, o, f, s in
```

Comment:
This basically does `x + 1.0` => `x + 1` which has been floating around in some cases and erroring compilation.

Reply:
That means that somewhere the offset is being considered/transformed as/to float?

Reply:
depending on staggered/x0 yes, where you can have like `x + .5*x.spacing` for example

Reply:
ok, thanks

Reply:
but this could cause internal reconstruction thus slowing down compilation. 

What's the cause of 1.0? I think we should rather intercept that and cast it to int

Reply:
> think we should rather intercept that and cast it to int

That's what it does, not sure what it would do better somewhere else, this is only doing it on the final index (i.e `x+1.0` which is tiny) rather than having to filter and `x + .5 * h_x` and figure out what to do with it and create Rationals, .....